saltproc/input_schema.json

```diff
@@ -63,7 +56,53 @@
                     "uniqueItems": false
                 }
             },
-            "required": ["codename", "exec_path", "template_inputfile_path", "iter_inputfile", "iter_matfile", "npop", "active_cycles","inactive_cycles", "geo_file_paths"]
+            "allOf": [
+                {
+                    "if": {
+                        "properties": { "codename": { "const": "serpent" } }
+                    },
+                    "then": {
+                        "properties": {
+                            "template_inputfiles_path": {
+                                "type": "string",
+                                "pattern": "^\\.\\/(.*)$"}
+                        }
+                    }
+                },
+                {
+                    "if": {
+                        "properties": { "codename": { "const": "openmc" } }
+                    },
+                    "then": {
+                        "properties": {
+                            "template_inputfiles_path": {
+                                "type": "object",
+                                "propterties": {
+                                    "settings": {
+                                        "type": "string",
+                                        "pattern": "^\\.\\/(.*)\\.xml$"},
+                                    "geometry": {
+                                        "type": "string",
+                                        "pattern": "^\\.\\/(.*)\\.xml$"},
+                                    "materials": {
+                                        "type": "string",
+                                        "pattern": "^\\.\\/(.*)\\.xml$"},
+                                    "plots": {
+                                        "type": "string",
+                                        "pattern": "^\\.\\/(.*)\\.xml$"},
+                                    "chain_file": {
+                                        "type": "string",
+                                        "pattern": "^\\.\\/(.*)\\.xml$"}
+                                },
+
+                                "required": ["settings", "geometry", "materials", "chain_file"]
```

Comment:
I'm not familiar with OpenMC, but how come it requires these terms while Serpent doesn't?

Reply:
You can write all of your Serpent input in a single file, but in practice this is cumbersome. Serpent uses the `include` card to link to other files in the main input file, which is useful when you have large material and/or geometry files. For Seprent, SaltProc assumes that the material and geometry files are included to the geometry file in this way.

OpenMC is different, in that it expects a different file for each input parameter. The input parameters here are `geometrt`, `materials`, `settings`, and `plots`. OpenMC depletion also requires a user-provided chain file, which is why we include it in `template_inputfiles_path` dictionary.